datarepo-clienttests/src/main/java/common/utils/KubernetesClientUtils.java

```diff
@@ -337,50 +339,66 @@ public static void changeReplicaSetSizeAndWait(int podCount) throws Exception {
       throw new RuntimeException("API deployment not found.");
     }
 
-    long apiPodCount = getApiPodCount(apiDeployment);
+    long apiPodCount = countPodsWithLabel(apiDeployment);
     logger.debug("Pod Count: {}; Message: Before scaling pod count", apiPodCount);
     apiDeployment = KubernetesClientUtils.changeReplicaSetSize(apiDeployment, podCount);
     KubernetesClientUtils.waitForReplicaSetSizeChange(apiDeployment, podCount);
 
     // print out the current pods
-    apiPodCount = getApiPodCount(apiDeployment);
+    apiPodCount = countPodsWithLabel(apiDeployment);
     logger.debug("Pod Count: {}; Message: After scaling pod count", apiPodCount);
-    printApiPods(apiDeployment);
+    logPodsWithLabel(apiDeployment);
   }
 
-  private static long getApiPodCount(V1Deployment deployment) throws ApiException {
+  /**
+   * Return a stream of the pods that match the deployment component label.
+   *
+   * @param deployment the deployment the pods belong to
+   * @return stream of matching pods
+   */
+  private static Stream<V1Pod> listPodsWithLabel(V1Deployment deployment) throws ApiException {
     String deploymentComponentLabel = deployment.getMetadata().getLabels().get(componentLabel);
     // loop through the pods in the namespace
     // find the ones that match the deployment component label (e.g. find all the API pods)
-    long apiPodCount =
-        listPods().stream()
-            .filter(
-                pod ->
-                    deploymentComponentLabel.equals(
-                        pod.getMetadata().getLabels().get(componentLabel)))
-            .count();
-    return apiPodCount;
+    return listPods().stream()
+        .filter(
+            pod ->
+                deploymentComponentLabel.equals(pod.getMetadata().getLabels().get(componentLabel)));
```

Comment:
Is there a chance `deploymentComponentLabel` could ever be null?

Reply:
I think probably yes. But that would mean that the Kubernetes cluster is in a wrong/unexpected state and we have to go investigate manually. So my thinking is that it doesn't really improves error reporting to add more null checks. (Also see earlier similar [comments](https://github.com/DataBiosphere/jade-data-repo/pull/725#discussion_r533448051).)

Reply:
cool